### PR TITLE
Migrate middleware to use edge config

### PR DESCRIPTION
**Pull request by busbyk**
See the new decision doc for details `docs/decisions/008-edge-config-tenant-lookup.md`.

The build-time generation was complicated and flawed due to the edge runtime. Edge config feels like a really nice solution and dang it's fast.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,54 +111,23 @@ To avoid an error when attempting to merge a PR on a feature branch into main, y
...
-- If database is available: generates fresh data from database
-- If database unavailable: uses `static-tenants.json` (committed baseline)
-- Final fallback: uses hardcoded data from `fallback-tenants.ts`
-
-### Updating Fallback Data
+We have two edge config stores in Vercel:
 
-When adding new tenants to production, update `src/utilities/tenancy/fallback-tenants.ts` to keep the baseline current. The seed file will automatically use the updated fallback data.
+1. For preview environment. This can also be used for local dev if needed for testing. But an edge config is not required for local development because of the cached API route fallback.
+2. For prod environment
```
Comment (rchlfryn) [inline README.md]: GH is not letting me select 2 lines like it normally does but ⛏️ 
```suggestion
1. For preview environment
2. For prod environment
NOTE: This can also be used for local dev if needed for testing, but an edge config is not required for local development because of the cached API route fallback.
```

**Review by rchlfryn — APPROVED** [review]
One minor edit - LG

File: README.md (hunk 1 above)
Comment (busbyk) [reply README.md]: I like it. Thanks!
